components/camel-tika/src/main/java/org/apache/camel/component/tika/TikaProducer.java

```diff
@@ -59,12 +59,24 @@
     private final String encoding;
 
     public TikaProducer(TikaEndpoint endpoint) {
+        this(endpoint, true, null);
+    }
+
+    public TikaProducer(TikaEndpoint endpoint, Parser parser) {
+        this(endpoint, parser == null, parser);
+    }
+
+    private TikaProducer(TikaEndpoint endpoint, boolean autodetectParser, Parser parser) {
         super(endpoint);
         this.tikaConfiguration = endpoint.getTikaConfiguration();
         this.encoding = this.tikaConfiguration.getTikaParseOutputEncoding();
         TikaConfig config = this.tikaConfiguration.getTikaConfig();
-        this.parser = new AutoDetectParser(config);
         this.detector = config.getDetector();
+        if (autodetectParser) {

```

Comment:
I think you can replace `autodetectParser` with `parser == null` and eliminate the three-args constructor.

Reply:
@ppalaga Unfortunately I can not do that, it will cause conflict with existing public constructor (https://github.com/apache/camel/pull/3705/files#diff-1383e03de9c8ef89d7546ab133b8801cR65). The only reason of having 3 args constructor is to keep property 'this.parser' final.

Reply:
I'll missed a small detail, I'll fix it in that way

Reply:
@ppalaga thanks for pointing this out. (I've over-combined so easy thing). It's changed now.